costmap_2d/CMakeLists.txt

```diff
@@ -74,7 +75,10 @@ catkin_package(
         roscpp
         sensor_msgs
         std_msgs
+        tf2
+        tf2_geometry_msgs
         tf2_ros
+        tf2_sensor_msgs
```

Comment:
this and tf2_geometry_msgs can't be CATKIN_DEPENDS, since they are declared as build dependencies only in the package.xml. This is because they are header-only packages, and those header files are only used in .cpp files (if they were leaked into our headers, then they become part of the public API, and we need to make them run depends as well as making them CATKIN_DEPENDS).